connectors/src/connectors/google_drive/temporal/spreadsheets.ts

```diff
@@ -420,39 +427,25 @@ export async function syncSpreadSheet(
           message: "Got 503 Service Unavailable from Google Sheets",
           type: "google_sheets_503_service_unavailable",
         };
-      }
-      throw err;
-    }
-  };
-
-  let internalErrorsCount = 0;
-  const maxInternalErrors = 3;
-  let spreadsheet: Awaited<ReturnType<typeof getSpreadsheet>>;
-  // If we consistently get 500 Internal Server Error from Google Sheets after 20 activity failures, we skip the file.
-  for (;;) {
-    try {
-      spreadsheet = await getSpreadsheet();
-      break;
-    } catch (err) {
-      if (err instanceof Error && "code" in err && err.code === 500) {
+      } else if (err instanceof Error && "code" in err && err.code === 500) {
```

Comment:
`nit` could be move to a typeguard like above.

Reply:
good point, will move to it in a separate PR, but have limited appetite for it right now (just need to unstuck)